src/util/stakeUtils.ts

```diff
@@ -84,8 +85,12 @@ export const getPolicyIconUris = (
     (rewardAssetName, i) => stakePolicy.rewardAssets[i].cdnName ?? metaTokens.find(metaToken => metaToken.currencyCode === rewardAssetName)?.contractAddress
   )
 
-  const stakeAssetUris = stakeContractAddresses.map(stakeContractAddress => getCurrencyIconUris(pluginId, stakeContractAddress ?? null).symbolImage)
-  const rewardAssetUris = rewardContractAddresses.map(rewardContractAddress => getCurrencyIconUris(pluginId, rewardContractAddress ?? null).symbolImage)
+  const stakeAssetUris = stakeContractAddresses.map(
+    stakeContractAddress => getCurrencyIconUris(pluginId, stakeContractAddress ?? null, SPECIAL_CURRENCY_INFO[pluginId]?.chainIcon ?? false).symbolImage
+  )
+  const rewardAssetUris = rewardContractAddresses.map(
+    rewardContractAddress => getCurrencyIconUris(pluginId, rewardContractAddress ?? null, SPECIAL_CURRENCY_INFO[pluginId]?.chainIcon ?? false).symbolImage
+  )
```

Comment:
This is pretty gnarly logic to keep two copies. Can we make a helper function?

```ts
const getAssetChainIcon = (contractAddress: string) =>
  getCurrencyIconUris(
    pluginId,
    contractAddress ?? null,
    // We want the chain icon if it makes sense:
    SPECIAL_CURRENCY_INFO[pluginId]?.chainIcon ?? false
  ).symbolImage

const stakeAssetUris = stakeContractAddresses.map(getAssetChainIcon)
const rewardAssetUris = rewardContractAddresses.map(getAssetChainIcon)
```

Reply:
Sure, will change it